### PR TITLE
[FIXED ROOT-35527] update to credentials 2.1.3

### PR DESCRIPTION
**Pull request by imod**
@stephenc could you have a quick look at this?

also, I'm not sure about `org.jenkinsci.plugins.configfiles.maven.security.CredentialsHelper.resolveCredentials(Item, List<ServerCredentialMapping>)` (it's used to resolve the credentials during a build) - should I adopt that too?

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/configfiles/maven/security/ServerCredentialMapping.java [hunk 1]
```diff
@@ -45,19 +52,27 @@ public String getCredentialsId() {
...
+        public ListBoxModel doFillCredentialsIdItems(@AncestorInPath ItemGroup context, @QueryParameter String serverId) {
+            AccessControlled _context = (context instanceof AccessControlled ? (AccessControlled) context : Jenkins.getInstance());
+            if (_context == null || !_context.hasPermission(Computer.CONFIGURE)) {
+                return new StandardUsernameListBoxModel().includeCurrentValue(serverId);
             }
-            return new ListBoxModel();
+
+            // @formatter:off
+            return new StandardUsernameListBoxModel().includeAs(
+                        ACL.SYSTEM,
```
Comment (stephenc) [inline src/main/java/org/jenkinsci/plugins/configfiles/maven/security/ServerCredentialMapping.java]: `context instanceof Queue.Task ? Tasks.getDefaultAuthenticationOf((Queue.Task)context) : ACL.SYSTEM` is probably what you want rather than a raw `ACL.SYSTEM`

**Comment by stephenc** [comment]
https://github.com/jenkinsci/config-file-provider-plugin/blob/master/src/main/java/org/jenkinsci/plugins/configfiles/maven/security/CredentialsHelper.java#L90 should *definately* be considering the `Tasks.getAuthenticationOf(item)`

Better yet is to migrate that method to use https://github.com/jenkinsci/credentials-plugin/blob/master/src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java#L817 as that should be the exact correct behaviour for resolving specific credentials in a specific build

**Comment by stephenc** [comment]
other than that, looking ok to the blind man on a galloping horse

**Comment by stephenc** [comment]
👍  AIUI

**Comment by imod** [comment]
@stephenc many thanks!

File: src/main/java/org/jenkinsci/plugins/configfiles/maven/security/ServerCredentialMapping.java [hunk 2]
```diff
@@ -45,19 +52,32 @@ public String getCredentialsId() {
...
     public static class DescriptorImpl extends Descriptor<ServerCredentialMapping> {
 
-        public ListBoxModel doFillCredentialsIdItems(@AncestorInPath AbstractProject<?, ?> context, @QueryParameter String serverId) {
-            if ( (context == null && Jenkins.getInstance().hasPermission(Jenkins.ADMINISTER))
-                    || (context != null && context.hasPermission(Item.CONFIGURE)) ) {
-                final List<StandardUsernameCredentials> validCredentials = CredentialsHelper.findValidCredentials(serverId);
-                return new StandardUsernameListBoxModel().withEmptySelection().withAll(validCredentials);
+        public ListBoxModel doFillCredentialsIdItems(@AncestorInPath ItemGroup context, @QueryParameter String serverId) {
+            AccessControlled _context = (context instanceof AccessControlled ? (AccessControlled) context : Jenkins.getInstance());
+            if (_context == null || !_context.hasPermission(Computer.CONFIGURE)) {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/configfiles/maven/security/ServerCredentialMapping.java]: `Computer.CONFIGURE` makes no sense here. Why did you change this from `Item.CONFIGURE`? Seems like a regression.

Comment (imod) [inline src/main/java/org/jenkinsci/plugins/configfiles/maven/security/ServerCredentialMapping.java]: fixed - thanks @jglick!
